### PR TITLE
Remove circular dependencies from our JavaScript code

### PR DESCRIPTION
**Pull request by gjvoosten**
Fix circular dependencies in our JavaScript code, and make introducing new circular dependencies an error now that we've fixed them all.

Closes NCI-Agency/anet#3320

### CODE REVIEW

File: client/config/webpack.common.js [hunk 1]
```diff
@@ -101,6 +102,17 @@ module.exports = {
...
         "process.env.NODE_ENV": JSON.stringify(process.env.NODE_ENV)
       }),
+      new CircularDependencyPlugin({
+        // exclude detection of files based on a RegExp
+        exclude: /node_modules/,
+        // add errors to webpack instead of warnings
+        failOnError: true,
+        // allow import cycles that include an asyncronous import,
+        // e.g. via import(/* webpackMode: "weak" */ './file.js')
+        allowAsyncCycles: false,
```
Comment (cemalettin-simsoft) [inline client/config/webpack.common.js:112]: this comment `// allow import cycles that include an asyncronous import,` and this value `allowAsyncCycles: false` seems contradicting ?

Comment (gjvoosten) [reply client/config/webpack.common.js:112]: I thought so too; it's a straight copy-paste from https://github.com/aackerman/circular-dependency-plugin#basic-usage though. ;-)

Comment (cemalettin-simsoft) [reply client/config/webpack.common.js:112]: Yeah, i checked for further explanations but didn't find any. Although, the library's code seems to be pointing to a contradiction as well. https://github.com/aackerman/circular-dependency-plugin/blob/master/index.js line:103
```
      // ignore dependencies that are resolved asynchronously
      if (this.options.allowAsyncCycles && dependency.weak) { continue }
```

Comment (gjvoosten) [reply client/config/webpack.common.js:112]: Unless you come to a definite conclusion of whether `false` actually means `true`, I'd say we leave it.

Comment (cemalettin-simsoft) [reply client/config/webpack.common.js:112]: Huh i got it, it means "Change this field if you want to allow it". That part is little confusing.

**Review by VassilIordanov — APPROVED** [review]

**Pull request merged**